### DISCUSSION TITLE
Using TabbedPage in MAUI application

### DISCUSSION THREAD

**Discussion by Orgbrat** [dotnet/maui#2569] [Q&A]
I need a TabbedPage in a MAUI Shell application (currently using WinUI for debugging). I  created  3 ContentPages and then created a TabbedPage to hold them as children as I did in Xamarin.Forms. But when I browse to the TabbedPage the first child displays but there are no Tabs on the bottom or top of the page.  I am not sure what I am doing wrong and I can not find a MAUI sample anywhere to checkout. Can anyone tell me what I might be doing wrong or direct me to a good and proper sample.

```
<TabbedPage x:Class="HealthManager.Pages.SettingsTabbedPage"
			 xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
			 xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
			 xmlns:local="clr-namespace:HealthManager.Pages"
			Title="Settings">
	<TabbedPage.Children>
		<local:SettingsGeneralPage />
		<local:SettingsSyncPage />
		<local:SettingsDataPage />
	</TabbedPage.Children>
</TabbedPage>
```

Orgbrat

**Comment by DanielRNichols** [comment]
Hi Orgbrat, Not sure if you ever found an answer to this, but I had the same problem. I found this video by James Montemango https://www.youtube.com/watch?v=YDxQjb7N-P8 and this article on .NET MAUI Preview 12  https://devblogs.microsoft.com/dotnet/announcing-net-maui-preview-12/

I was able to achieve what I was trying to do by modifying the AppShell:

    <TabBar>
        <Tab Icon="dotnet_bot.png" Title="Home">
            <ShellContent ContentTemplate="{DataTemplate local:HomePage}"/>
        </Tab>
        <Tab Icon="test.png" Title="Tab2">
            <ShellContent Route="Page1" ContentTemplate="{DataTemplate local:Page1}"/>
            <ShellContent Route="Page2" ContentTemplate="{DataTemplate local:Page2}"/>
            <ShellContent Route="Page3" ContentTemplate="{DataTemplate local:Page3}"/>
        </Tab>
    </TabBar>

**Reply by Awesomer9561** [reply]
Hi, you have done this using shell, but what non-shell pages? Tabbed page is not rendering top tabs, just the content is swiping.

**Reply by Dmaui** [reply]
I wanted to use Tabs without shell. In Xamarin Forms we can write the renderer and can completely customize the tab page the way we wanted. I'm not sure how we can use Handlers. Mostly handlers samples and the other documentation is talking about color changes or text edits. I never found a detailed documentation on how to customize a page renderer like we used to do in Xamarin. Mostly we should have the capability to do this in MAUI or else the upgrade is actually stopping to do more native usecases. Can somebody provide some detailed link on how to customize pages and controls to use more native features or native libs rather than writing everything cross platorm.
